eoc_journal/api_client.py

```diff
@@ -0,0 +1,113 @@
+"""
+An XBlock that allows learners to download their activity after they finish their course.
+"""
+
+from django.conf import settings
+
+import requests
+
+
+def calculate_engagement_score(social_metrics):
+    """
+    Calculates and returns the total score for the given social metrics
+    """
+    engagement_total = 0
+
+    for key, val in _get_social_metrics_points().iteritems():
+        engagement_total += social_metrics.get(key, 0) * val
+    return engagement_total
+
+
+def _get_edx_api_key():
+    """
+    Returns the EDX_API_KEY from the django settings.
+
+    If key is not set, returns None.
```

Comment:
Please mention here that this API key must never be sent to the client or used in a template, as it's for server-to-server use only.